### PR TITLE
Create boot directory if not already declared.

### PR DESCRIPTION
**Pull request by tuxmea**
Fixes GH-580

### CODE REVIEW

File: manifests/plugin/discovery.pp [hunk 1]
```diff
@@ -25,6 +25,12 @@
   if $install_images {
     $tftp_root_clean = regsubst($tftp_root, '/$', '')
 
+    if ! defined(File["${tftp_root_clean}/boot"]) {
```
Comment (ekohl) [inline manifests/plugin/discovery.pp]: I prefer ```ensure_resource('file', "${tftp_root_clean}/boot", {ensure => directory})```

Comment (tuxmea) [reply manifests/plugin/discovery.pp]: Done. I only wonder why this is now three commits...

Comment (mmoll) [reply manifests/plugin/discovery.pp]: I think this is still not correct. ;)
Please squash all your commits together and force-push, if possible - however, we can also fix this up on merge...

File: manifests/plugin/discovery.pp [hunk 2]
```diff
@@ -25,6 +25,8 @@
   if $install_images {
     $tftp_root_clean = regsubst($tftp_root, '/$', '')
 
+    ensure_resource('file', "${tftp_root_clean}/boot", {ensure => directory})
+
     foreman::remote_file {"${tftp_root_clean}/boot/${image_name}":
       remote_location => "${source_url}${image_name}",
       mode            => '0644',
```
Comment (mmoll) [inline manifests/plugin/discovery.pp]: I think this should require that file resource then.

Comment (tuxmea) [reply manifests/plugin/discovery.pp]: Which is the oldest Puppet version supported by Foreman? Puppet 3? In Puppet 4 we have ordering = manifest as default option set.

Comment (mmoll) [reply manifests/plugin/discovery.pp]: Hrm, that would be 4.6.2.

Comment (ekohl) [reply manifests/plugin/discovery.pp]: Personally I prefer not to rely on manifest ordering

Comment (tuxmea) [reply manifests/plugin/discovery.pp]: OK. Done.

**Pull request closed without merging**